proposals/005_Experimental.md

```diff
@@ -0,0 +1,115 @@
+
+### Schema Object
+
+...
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+... | ... | ...
+<a name="schemaExperimental"></a>experimental | `boolean` | Specifies that a schema is in experimental status, meaning it may change outside of the normal breaking change process. Consumers SHOULD use with caution. Default value is `false`.
+
+### Example Spec
+
+```yaml
+  /asset/constraints:
+    get:
+      tags:
+        - Asset
+        - Constraints
+      summary: Get a set of asset constraints
+      operationId: constraints
+      parameters:
+        - name: siteToken
+          in: query
+          description: Site token obtained from Site API
+          required: true
+          schema:
+            type: string
+      experimental: true
+```
+### Prior Art
+
+This kind of requirement is handled for TypeScript libraries by [api-extractor](https://api-extractor.com/pages/tsdoc/doc_comment_syntax/#release-tags) - they have both "alpha" and "beta" markers with a somewhat opinionated flow attached - I'm not sure that level of granularity is necessary. But the "beta" and "public" ones map well to the motivations described here:
+
+> - **beta**: Indicates that an API item has been released as a preview or for experimental purposes. Third parties are encouraged to try it and provide feedback. However, a “beta” API should NOT be used in production, because it may be changed or removed in a future version.
+> - **public**: Indicates that an API item has been officially released, and is now part of the supported contract for a package. If the SemVer versioning scheme is used, then the API signature cannot be changed without a MAJOR version increment.
+
+### Unanswered Questions
+
+- If an operation is not marked as experimental, but it is using a schema which is (i.e. as its request object), then it is implicitly also unstable. Would this usage be considered invalid?
+- Should `experimental` and `deprecated` be mutually exclusive?
+
+## Backwards compatibility
+
+The `experimental` field would default to false, meaning existing behaviour is preserved, and the new field is only used on an opt-in basis.
+
+## Alternatives considered
+
+- _Specification extensions_ - publishers could add an extension in their own domain, but the benefit of the metadata being available to downstream tools (including those used by consumers, not just publishers) would be lost.
+- _Tags_ - as above, but this also gets to mixing other kinds of metadata in with resource taxonomy, which seems wrong.
+- _Different API_ - this would be the least messy from a technical perspective - maintain a completely separate API for experimental items, and then "promote" them to the main API once they are considered stable. This has increased overhead for publishers and consumers, and could also reduce the likelihood of getting feedback on, and early uptake of, experimental items if they are segregated in a different place altogether.
+
```

Comment:
Can we add a note about use of [overlays](https://github.com/OAI/OpenAPI-Specification/blob/master/proposals/004_Overlays.md#overlays) as a potential alternative? A semver-versioned base OAS document with one-or-more overlays labelled 'experimental' might meet the use-case too.

Reply:
Overlays looks very interesting! I do think it could work although perhaps more complex to implement against. Added an item to the alternatives section.